sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -809,6 +809,7 @@ class MultiFileParquetPartitionReader(
         if (!isCorrectRebaseMode) {
           (0 until table.getNumberOfColumns).foreach { i =>
             if (RebaseHelper.isDateTimeRebaseNeededRead(table.getColumn(i))) {
+              table.close()
```

Comment:
nit: can we use withResource for table as well?

Reply:
Seems like `closeOnExcept` is more appropriate here.